### PR TITLE
[Intel MKL] Adding support for MKL builds with AVX2. 

### PR DESCRIPTION
**Pull request by claynerobison**
MKL-DNN ignores the compiler switches and takes code paths based on the platform detected at runtime. This commit will add support for avx2 instructions to the rest of TensorFlow.

### CODE REVIEW

File: tensorflow/tools/ci_build/linux/mkl/build-dev-container.sh [hunk 1]
```diff
@@ -49,5 +52,27 @@ TF_DOCKER_BUILD_TYPE="MKL" \
   TF_DOCKER_BUILD_IMAGE_NAME="${TF_DOCKER_BUILD_IMAGE_NAME}" \
   TF_DOCKER_BUILD_VERSION="${TF_DOCKER_BUILD_VERSION}" \
   TF_DOCKER_BUILD_PYTHON_VERSION="PYTHON3" \
+  ${WORKSPACE}/tensorflow/tools/docker/parameterized_docker_build.sh
+
+# Build containers for AVX2
+TF_BAZEL_BUILD_OPTIONS="--config=mkl --copt=-mavx2 --cxxopt=-D_GLIBCXX_USE_CXX11_ABI=0"
```
Comment (rmlarsen) [inline tensorflow/tools/ci_build/linux/mkl/build-dev-container.sh]: Does AVX2 imply FMA and AVX? The two latter are typically more important for performance. Should we explicitly list all three?

File: tensorflow/tools/ci_build/linux/mkl/build-dev-container.sh [hunk 2]
```diff
@@ -34,6 +34,9 @@ echo "TF_DOCKER_BUILD_DEVEL_BRANCH=${TF_DOCKER_BUILD_DEVEL_BRANCH}"
 echo "TF_DOCKER_BUILD_IMAGE_NAME=${TF_DOCKER_BUILD_IMAGE_NAME}"
 echo "TF_DOCKER_BUILD_VERSION=${TF_DOCKER_BUILD_VERSION}"
 
+# Build containers for AVX
+#"TF_BAZEL_BUILD_OPTIONS": "'{}'" (default build option= avx)
```
Comment (rmlarsen) [inline tensorflow/tools/ci_build/linux/mkl/build-dev-container.sh]: Cruft?

**Review by rmlarsen — CHANGES_REQUESTED** [review]

**Comment by claynerobison** [comment]
@rmlarsen Clarified comments and added more complete support for the haswell and sandybridge instruction sets. The mtune parameters won't add instructions, but will tune cache sizes etc for the specified architecture.

**Review by rmlarsen — DISMISSED** [review]

**Comment by rmlarsen** [comment]
@claynerobison Thanks!

**Comment by rmlarsen** [comment]
@gunan @jhseu does this look good to you?

**Review by gunan — DISMISSED** [review]

**Review by jhseu — DISMISSED** [review]

**Comment by claynerobison** [comment]
Sorry, @jhseu @rmlarsen @gunan further internal testing revealed that we didn't have the full haswell arch build flag being passed in to the first two containers. Fixed it with the last commit.

**Review by rmlarsen — APPROVED** [review]

**Review by gunan — APPROVED** [review]

**Pull request merged**